x-pack/plugins/apm/server/routes/settings/agent_configuration/find_exact_configuration.ts

```diff
@@ -59,11 +56,33 @@ export async function findExactConfiguration({
     return;
   }
 
+  const appliedByAgent = await getIsAppliedByAgent({
+    apmEventClient,
+    agentConfiguration: hit._source,
+  });
+
   return {
     id: hit._id,
     ...convertConfigSettingsToString(hit)._source,
-    applied_by_agent:
-      hit._source.applied_by_agent ||
-      configsAppliedToAgentsThroughFleet.hasOwnProperty(hit._source.etag),
+    applied_by_agent: appliedByAgent,
   };
 }
+
+async function getIsAppliedByAgent({
```

Comment:
You could probably create a simple unit test to cover this function.

Reply:
This is already covered by the api test. A unit test would mock out the call to `getAgentConfigEtagMetrics` so it would only cover the check on `agentConfiguration.applied_by_agent` and `appliedEtags.includes(agentConfiguration.etag`. I don't think it's worth it.